homeassistant/components/mqtt/client.py

```diff
@@ -208,12 +220,15 @@ async def async_subscribe(
             translation_domain=DOMAIN,
             translation_placeholders={"topic": topic},
         ) from exc
-    return await mqtt_data.client.async_subscribe(
-        topic,
-        msg_callback,
-        qos,
-        encoding,
-    )
+    client = mqtt_data.client
+    if not client.connected and not mqtt_config_entry_enabled(hass):
```

Comment:
Should the case not be:

```suggestion
    if not client.connected or not mqtt_config_entry_enabled(hass):
```

Reply:
Since the client may be reconnecting I think we have to do both

Reply:
Good point

Reply:
It looks like we queue subscriptions if we aren't connected so I think its correct
https://github.com/home-assistant/core/blob/569763b7a83411eecfdd1a9f60c4ab7f13bd68b6/homeassistant/components/mqtt/client.py#L875